### PR TITLE
GORA-548 Apache Flink support for Gora

### PR DESCRIPTION
**Pull request by djkevincr**
Added the initial implementation with this PR.

### CODE REVIEW

File: gora-core/pom.xml [hunk 1]
```diff
@@ -171,6 +187,12 @@
       <artifactId>hadoop-minicluster</artifactId>
       <scope>test</scope>
     </dependency>
+      <dependency>
```
Comment (lewismc) [inline gora-core/pom.xml]: Why is this in here?

Comment (djkevincr) [reply gora-core/pom.xml]: +1

**Comment by djkevincr** [comment]
@lewismc Yes implementation really simple, since Flink has this Hadoop Input format wrappers so we can reuse the Gora Input formats directly as data sources and sinks to Apache Flink jobs. I will write common common test code similar to what we have on MapReduceTestUtils so that people write test cases based on Flink Jobs. Documentation will follow.

**Comment by djkevincr** [comment]
I am currently working on writing something similar to MapReduceTestUtils so that people write test cases based on Flink Jobs.

**Comment by lewismc** [comment]
```
[ERROR] Failures:
[ERROR]   TestHBaseStore>DataStoreTestBase.testDelete:341 expected:<9> but was:<8>
[ERROR]   TestHBaseStore>DataStoreTestBase.testDeleteByQuery:347 expected:<10> but was:<9>
[ERROR]   TestHBaseStore>DataStoreTestBase.testDeleteByQueryFields:353 expected:<10> but was:<9>
[ERROR]   TestHBaseStore>DataStoreTestBase.testGetWebPageDefaultFields:287
[ERROR]   TestHBaseStore>DataStoreTestBase.testQueryEndKey:311 expected:<9> but was:<8>
[ERROR]   TestHBaseStore>DataStoreTestBase.testQueryKeyRange:317 expected:<10> but was:<9>
[ERROR]   TestHBaseStore>DataStoreTestBase.testQueryStartKey:305 expected:<10> but was:<9>
[ERROR]   TestHBaseStore>DataStoreTestBase.testQueryWebPageSingleKey:323
[ERROR]   TestHBaseStore>DataStoreTestBase.testQueryWebPageSingleKeyDefaultFields:329
[ERROR]   TestHBaseStore>DataStoreTestBase.testResultSizeKeyRangeWithLimit:407 expected:<0> but was:<1>
[ERROR] Errors:
[ERROR] org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount.testFlinkWordCountFlink(org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount)
[ERROR]   Run 1: TestHBaseStoreWordCount.setUp:44 » SocketTimeout callTimeout=1200000, callDura...
[ERROR]   Run 2: TestHBaseStoreWordCount.tearDown:53 NullPointer
[INFO]
[INFO]
[ERROR] Tests run: 53, Failures: 10, Errors: 1, Skipped: 4
```

**Comment by lewismc** [comment]
[org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount.txt](https://github.com/apache/gora/files/3027131/org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount.txt)
[org.apache.gora.hbase.store.TestHBaseStore.txt](https://github.com/apache/gora/files/3027132/org.apache.gora.hbase.store.TestHBaseStore.txt)

Test logs attached

**Comment by djkevincr** [comment]
Test passes without any issue in my machine locally for HBase, I will have a another look.

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.gora.hbase.util.TestHBaseByteInterface
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.851 s - in org.apache.gora.hbase.util.TestHBaseByteInterface
[INFO] Running org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount
Formatting using clusterid: testClusterID
[WARNING] Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 62.243 s - in org.apache.gora.hbase.mapreduce.TestHBaseStoreWordCount
java.lang.ThreadGroup[name=PEWorkerGroup,maxpri=10]
    Thread[IPC Client (1234715829) connection to localhost/127.0.0.1:41613 from djkevincr,5,PEWorkerGroup]
[INFO] Running org.apache.gora.hbase.mapreduce.TestHBaseStoreMapReduceSerialization
Formatting using clusterid: testClusterID
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.424 s - in org.apache.gora.hbase.mapreduce.TestHBaseStoreMapReduceSerialization
[INFO] Running org.apache.gora.hbase.mapreduce.TestHBaseStoreCountQuery
Formatting using clusterid: testClusterID
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 19.039 s - in org.apache.gora.hbase.mapreduce.TestHBaseStoreCountQuery
[INFO] Running org.apache.gora.hbase.store.TestHBaseStore
Formatting using clusterid: testClusterID
[WARNING] Tests run: 46, Failures: 0, Errors: 0, Skipped: 3, Time elapsed: 121.957 s - in org.apache.gora.hbase.store.TestHBaseStore
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 53, Failures: 0, Errors: 0, Skipped: 4

**Comment by djkevincr** [comment]
I will monitor over this @lewismc, I sometime notice this intermittent failures with HBase. I will address that separately.

**Pull request merged**